packages/dataclass-editor/src/components/dataclass/data/dataclass-utils.ts

```diff
@@ -10,6 +21,19 @@ export const classTypeOf = (dataClass: DataClass) => {
   return 'DATA';
 };
 
+export const isEntity = (
+  dataClass: DataClass
+): dataClass is DataClass & { entity: DataClassEntity; fields: Array<DataClassField & { entity: DataClassFieldEntity }> } => {
+  return !!dataClass.entity;
+};
+
+export const isEntityField = (field?: DataClassField): field is DataClassField & { entity: DataClassFieldEntity } => {
+  if (!field) {
+    return false;
+  }
+  return !!field.entity && field.modifiers.includes('PERSISTENT');
+};
```

Comment:
I feel like the management of the whole changing of the UI based on whether the data class is an entity or not is lacking. I am using these functions in places that should realistically only be reachable if the data class is an entity but it needs to be rechecked to make TypeScript happy. Any suggestions on determining this flag better on a global scale?

Reply:
I thought about extending the data class type and making the `entity` property not optional but since it gets accessed through the app context I don't think that would work well.

Reply:
understandable. 
Hm I didn't read the code yet, so hard to tell. But yes normally this is only possible through generics or via these type guards...
But instead of use this function everywhere, why not have a second attribute in the context hook which delivers a entyitydataclass? Or maybe even better, create a new useEntitiyDataClass hock which uses internally the data class from the context but with this check.